threads/test_thread_rand.c

```diff
@@ -0,0 +1,88 @@
+#include <string.h>
+#include <stdlib.h>
+#include <unistd.h>
+#include <sys/threads.h>
+
+#include "unity_fixture.h"
+
+
+typedef struct {
+	uint16_t time;
+	uint16_t id;
+} threads_self_t;
+
+
+static int cthread = 10, fin = 0;
+
+
+static char stack[10 * 512];
+
+
+static void test_thread(void *arg)
+{
+	threads_self_t *self = (threads_self_t *)arg;
+	usleep(1000 * self->time);
+	__sync_fetch_and_add(&fin, 1);
+	endthread();
+}
+
+
+TEST_GROUP(threads_rand);
+
+
+TEST_SETUP(threads_rand)
+{
+}
+
+
+TEST_TEAR_DOWN(threads_rand)
+{
+}
+
+
+TEST(threads_rand, test_1)
+{
+	threads_self_t params[cthread];
+	int ret, joined = 0, i;
+
+	srand(7);
+	/* assign threads parameters */
+	for (i = 0; i < cthread; i++) {
+		params[i].id = i;
+		params[i].time = 100 + rand() % 200; /* assign thread lifespan in milliseconds */
+	}
+
+	/* run threads */
+	for (i = 0; i < cthread; i++) {
+		beginthread(test_thread, 6, &(stack[i * 512]), 512, &(params[i]));
+		usleep(100);
```

Comment:
What is the reason of adding here 100us delay?

Reply:
Yes it is unnecessary here. Removed.